libs/utils/alloc_array.h

```diff
@@ -49,25 +50,15 @@ namespace utils
 
         alloc_array(const vtype new_value)
         {
+  	    auto new_value_tmp = move(new_value);
```

Comment:
C'est vraiment utile ça 🤔

Reply:
si on a un type qui n'a pas de copie

Reply:
Si on move pas `new_value` c'est une copie en trop. A la limite je peux surcharger la fonction pour différencier les lvalues des rvalues...

Reply:
Je ne vois pas un cas où tu veux remplir une array avec un type qui n'est pas copyable

Reply:
un élément avec un unique_ptr initialisé en nullptr peut être ? le type ne peut être copyable par défaut

Reply:
Un autre truc c'est que dans ton array touts les éléments ne sonts pas directement les un a coté des autres, genre ça ressemble plus a un vecteur avec un taille max définie

Reply:
ow j'ai pas pensé à ça :( sorry

Reply:
Une implémentation de `unique_ptr`, par exemple.
Sinon, [https://sirlynixvanfrietjes.be/2018/02/25/cpp-moderne-le-passage-dobjets-par-argument/](un article pour expliquer la démarche.)

Reply:
Sinon ouais, ton impl est un peu tordue x)

Reply:
Merde, j'ai pas pris en compte le fait qu'il y avait une boucle dans ton ctor, en effet, le `move` ne sert à rien.